rotkehlchen/chain/ethereum/modules/juicebox/decoder.py

```diff
@@ -0,0 +1,84 @@
+import logging
+from typing import Any
+
+from rotkehlchen.chain.ethereum.abi import decode_event_data_abi_str
+from rotkehlchen.chain.ethereum.modules.juicebox.constants import (
+    CPT_JUICEBOX,
+    PAY_ABI,
+    PAY_SIGNATURE,
+    TERMINAL_3_1_2,
+)
+from rotkehlchen.chain.ethereum.utils import token_normalized_value_decimals
+from rotkehlchen.chain.evm.decoding.interfaces import DecoderInterface
+from rotkehlchen.chain.evm.decoding.structures import (
+    DEFAULT_DECODING_OUTPUT,
+    DecoderContext,
+    DecodingOutput,
+)
+from rotkehlchen.chain.evm.decoding.types import CounterpartyDetails
+from rotkehlchen.constants.assets import A_ETH
+from rotkehlchen.errors.serialization import DeserializationError
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import ChecksumEvmAddress
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class JuiceboxDecoder(DecoderInterface):
+
+    def _decode_pay(self, context: DecoderContext) -> DecodingOutput:
+        """Decode pay with rewards in juicebox"""
+        if context.tx_log.topics[0] != PAY_SIGNATURE:
+            return DEFAULT_DECODING_OUTPUT
+
+        try:
+            _, decoded_data = decode_event_data_abi_str(context.tx_log, PAY_ABI)
+        except DeserializationError as e:
+            log.error(
+                f'Failed to deserialize Juicebox event at '
+                f'{context.transaction.tx_hash.hex()} due to {e}',
+            )
+            return DEFAULT_DECODING_OUTPUT
+
+        amount = token_normalized_value_decimals(
+            token_amount=decoded_data[2],
+            token_decimals=18,
+        )
+
+        for event in context.decoded_events:
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.asset == A_ETH and
+                event.balance.amount == amount and
+                event.location_label == decoded_data[0]
+            ):
+                event.counterparty = CPT_JUICEBOX
+                event.notes = f'Payed {amount} ETH at Juicebox'
+                if len(decoded_data[4]) != 0:
+                    event.notes += f' with memo: "{decoded_data[4]}"'
+            elif (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE
+            ):
+                event.counterparty = CPT_JUICEBOX
+                event.notes = 'Received reward after payment in Juicebox'
```

Comment:
What is this? Reward is a strong word. it's always just a small nft to commemorate the donation no? 

```suggestion
                event.notes = 'Received a commemorative NFT for donating via Juicebox'
```

If it's always an NFT